redhat-cop/automate_tower_genie_collections
# Added option to not removing $encrypted$ from filetree_create exported credentials

# What does this PR do?
Added option to not removing `$encrypted$` from filetree_create exported credentials.  It keeps the `$encrypted$` text, instead of `` empty string, in the credential files.  It prevents the existing credential value from being overwritten with empty string if the configure files are being used as-is to import back into AAP.


A condition is added to roles/filetree_create/templates/current_credentials.j2 to control the output.

# How should this be tested?
Include `show_encrypted: true` in task invoking `filetree_create`
```yaml
- name: Export AAP credentials for organization {{ org_name }}
  ansible.builtin.import_role:
    name: infra.controller_configuration.filetree_create
  vars:
    organization_filter: "{{ org_name }}"
    output_path: "{{ _objects_output_dir }}"
    flatten_output: false
    input_tag:
      - credentials
    show_encrypted: true
```
Observe password fileds in output files 
```yaml
---
controller_credentials:
  - name: "SSH root"
    description: "Test"
    credential_type: "Machine"
    organization: "Default"
    inputs:
      become_method: ''
      become_username: ''
      ssh_key_data: $encrypted$
      ssh_key_unlock: $encrypted$
      username: root
```


# Is there a relevant Issue open for this?


resolves #788

# Other Relevant info, PRs, etc
N/A
